protocol/messenger.go

```diff
@@ -4040,24 +4051,68 @@ func (m *Messenger) MessageByChatID(chatID, cursor string, limit int) ([]*common
 
 	if m.httpServer != nil {
 		for idx := range msgs {
-			m.prepareMessage(msgs[idx], m.httpServer)
+			err = m.prepareMessage(msgs[idx], m.httpServer)
+
+			if err != nil {
+				return nil, "", err
+			}
 		}
 	}
 
 	return msgs, nextCursor, nil
 }
 
-func (m *Messenger) prepareMessages(messages map[string]*common.Message) {
+func (m *Messenger) prepareMessages(messages map[string]*common.Message) error {
 	if m.httpServer != nil {
 		for idx := range messages {
-			m.prepareMessage(messages[idx], m.httpServer)
+			err := m.prepareMessage(messages[idx], m.httpServer)
+
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func extractQuotedImages(messages []*common.Message, s *server.MediaServer) []string {
+	var quotedImages []string
+
+	for _, message := range messages {
+		if message.ChatMessage != nil && message.ChatMessage.ContentType == protobuf.ChatMessage_IMAGE {
+			quotedImages = append(quotedImages, s.MakeImageURL(message.ID))
 		}
 	}
+	return quotedImages
 }
 
-func (m *Messenger) prepareMessage(msg *common.Message, s *server.MediaServer) {
+func (m *Messenger) prepareMessage(msg *common.Message, s *server.MediaServer) error {
 	if msg.QuotedMessage != nil && msg.QuotedMessage.ContentType == int64(protobuf.ChatMessage_IMAGE) {
 		msg.QuotedMessage.ImageLocalURL = s.MakeImageURL(msg.QuotedMessage.ID)
```

Comment:
can we not pass `s *server.MediaServer` and just verify it from this fn?
```
func (m *Messenger) prepareMessage(msg *common.Message, s *server.MediaServer) error {
    if m.httpServer == nil {
        return nil
    }
...
}
```

Reply:
```go
func (m *Messenger) prepareMessage(msg *common.Message, s *server.MediaServer) error {
    if m.httpServer == nil {
        return nil
    }
...
```

Not really sure I understand the logic in that code snippet you provided. 
Furthermore, if the intent is to check if provided parameter is nil, my take on this issue is that the caller function should already have had that parameter as nil and should have checked that the parameter is nil.
The reason why I advocate for this is because, let's say there is bug in the function and this produces a bug. As developer, I want to fix the root cause, not the function down the road that processes the parameter. You can see the check example here :

```go
...
	if m.httpServer != nil {
		for idx := range msgs {
			err = m.prepareMessage(msgs[idx], m.httpServer)

			if err != nil {
				return nil, "", err
			}
		}
	}
...
```
https://github.com/status-im/status-go/blob/405d468e0e74e78a89f0653ad6514c50af9d1d6f/protocol/messenger.go#L4054

Reply:
I mean, that we don't need to pass m.httpServer all over the way as m.httpServer is a member of the Messenger class

Reply:
Oh I see. Good catch 👍🏾 . I think that can be part of a refactoring I guess. I can open a ticket for this cause it'll change a bunch of files all over the places.